nearswap/tests/fee_simulation.rs

```diff
@@ -0,0 +1,282 @@
+            nearswap.get_deposit_token(alice.account_id, dai())
+        ).unwrap_json::<U128>();
+
+    assert_eq!(
+        to_u128(before_swap_token) + to_u128(price_n2t),
+        to_u128(after_swap_token), "Near to token swap unsuccessful");
+
+    // Check if fee is deducted - Near deposited into pool must be 0.997 * near amount
+    // Fee - 0.3%
+    let out = mock_calc_out(
+        to_yocto("5")*997/1000,
+        to_u128(pool_before.ynear), to_u128(pool_before.tokens)
+    );
+    assert_eq!(out, to_u128(after_swap_token), "Wrong amount of fee deducted");
+
+    pool_before = view!(nearswap.pool_info(&dai())).unwrap_json::<PoolInfo>();
+    let before_swap_token_lp1 = view!(
+        nearswap.get_deposit_token(lp1.account_id.clone(), dai())
+    ).unwrap_json::<U128>();
+    let before_swap_near_lp1 = view!(
+        nearswap.get_deposit_near(lp1.account_id.clone())
+    ).unwrap_json::<U128>();
+    let before_swap_token_lp2 = view!(
+        nearswap.get_deposit_token(lp2.account_id.clone(), dai())
+    ).unwrap_json::<U128>();
+    let before_swap_near_lp2 = view!(
+        nearswap.get_deposit_near(lp2.account_id.clone())
+    ).unwrap_json::<U128>();
+
+    // withdraw liquidity
+    call!(
+        lp1,
+        nearswap.withdraw_liquidity(dai(), lp1_shares, U128(1), U128(1))
+    ).assert_success();
+
+    let pool_after = view!(nearswap.pool_info(&dai())).unwrap_json::<PoolInfo>();
+    let after_swap_token_lp1 = view!(
+        nearswap.get_deposit_token(lp1.account_id.clone(), dai())
+    ).unwrap_json::<U128>();
+    let after_swap_near_lp1 = view!(
+        nearswap.get_deposit_near(lp1.account_id.clone())
+    ).unwrap_json::<U128>();
+
+    // Check If ~90% of total shares are received by lp1
+    let tokens_received_lp1 = to_u128(after_swap_token_lp1) - to_u128(before_swap_token_lp1);
+    assert_eq!(to_u128(pool_before.tokens)*9/10, tokens_received_lp1, "Redeemed liquidity is not correct for lp1");
+
+    let near_received_lp1 = to_u128(after_swap_near_lp1) - to_u128(before_swap_near_lp1);
+    assert_eq!(to_u128(pool_before.ynear)*9/10, near_received_lp1, "Redeemed Near incorrect - lp1");
+    
+    call!(
+        lp2,
+        nearswap.withdraw_liquidity(dai(), lp2_shares, U128(1), U128(1))
+    ).assert_success();
+
+    let after_swap_token_lp2 = view!(
+        nearswap.get_deposit_token(lp2.account_id.clone(), dai())
+    ).unwrap_json::<U128>();
+    let after_swap_near_lp2 = view!(
+        nearswap.get_deposit_near(lp2.account_id.clone())
+    ).unwrap_json::<U128>();
+
+    // Check If ~10% of total shares are received by lp2
+    let tokens_received_lp2 = to_u128(after_swap_token_lp2) - to_u128(before_swap_token_lp2);
+    assert_close(U128(to_u128(pool_before.tokens)*1/10), tokens_received_lp2, 1);
+
+    let near_received_lp2 = to_u128(after_swap_near_lp2) - to_u128(before_swap_near_lp2);
+    assert_eq!(to_u128(pool_before.ynear)*1/10, near_received_lp2, "Redeemed Near incorrect - lp2");
```

Comment:
let's also check that received NEAR = 1 (initial deposit) + 5*0.3% * 1/0  (the fee - LP has 10% of the share, so 10% of the fee should go to him.)

Reply:
ok

Reply:
but according to pool, after swap it has 15 Near therefore these accounts get 
10% and 90% share. i.e 1.5 and 13.5 near. how do i check fees in this ?

Reply:
we checked that alice is getting tokens for 0.997*input_amount, and 0.3% is kept in pool.

Reply:
right,  so, alice is getting less tokens and in pool  we have 15NEAR.
So let's check that before withdraws, in the pool we have:
```
pool_after_swap.tokens > pool_before_swap.tokens - alice_tokens_out
```

Reply:
i think this condition will be equal because we checking tokens only

Reply:
ah, you are right. Hmm, the fee is hidden in the `alice_tokens_out` -- she receive less than what she pays. But it's hard to check. I think the only test we can do is a unit-test for `calc_out_with_fee` that it's smaller than `calc_out_amount`

Reply:
yes, added a test for it.